src/Cli.Tests/UtilsTests.cs

```diff
@@ -234,6 +234,64 @@ public void TestMergeConfig()
             }
         }
 
+        /// <summary>
+        /// Test to verify that merged config file is only used for the below scenario
+        /// 1. Environment value is set.
+        /// 2. Both Base and envBased config file is present.
+        /// In all other cases, the TryMergeConfigsIfAvailable method should return false
+        /// and out string for the mergedConfigFile should be empty.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow("", false, false, "", false, DisplayName = "If environment value is not set, merged config file is not generated.")]
+        [DataRow("", false, true, "", false, DisplayName = "If environment value is not set, merged config file is not generated.")]
+        [DataRow("", true, false, "", false, DisplayName = "If environment value is not set, merged config file is not generated.")]
+        [DataRow("", true, true, "", false, DisplayName = "If environment value is not set, merged config file is not generated.")]
+        [DataRow("Test", false, false, "", false, DisplayName = "Environment value set but base config not available, merged config file is not generated.")]
+        [DataRow("Test", false, true, "", false, DisplayName = "Environment value set but base config not available, merged config file is not generated.")]
+        [DataRow("Test", true, false, "", false, DisplayName = "Environment value set but env based config not available, merged config file is not generated.")]
+        [DataRow("Test", true, true, "dab-config.Test.merged.json", true, DisplayName = "Environment value set and both base and envConfig available, merged config file is generated.")]
+        public void TestMergeConfigAvailability(
```

Comment:
Was there no other test testing these scenarios indirectly ie End 2 End?

Reply:
Nope, Last time the test was focused more the merge capabilities.